src/compute/src/server.rs

```diff
@@ -381,6 +381,12 @@ impl<'w, A: Allocate + 'static> Worker<'w, A> {
                                 };
                             }
                             cap.downgrade(&(time + 1));
+                        } else {
+                            // Non-leader workers will still receive `UpdateConfiguration` commands
+                            // and we must drain those to not leak memory.
+                            if let Ok(_cmd) = command_rx.try_recv() {
+                                assert_ne!(worker_id, 0);
+                            }
```

Comment:
This is potentially fragile. Could we ensure that `cmd` is only ever an `UpdateConfiguration` variant? Otherwise, we might be confused why things don't behave as expected should we ever change how we distribute commands from the controller to replicas.

Reply:
Added a second assert for the command variant.